ios/FluentUI/Badge Field/BadgeView.swift

```diff
@@ -395,10 +395,13 @@ open class BadgeView: UIView, TokenizedControlInternal {
                                                name: UIContentSizeCategory.didChangeNotification,
                                                object: nil)
 
-        NotificationCenter.default.addObserver(self,
-                                               selector: #selector(themeDidChange),
-                                               name: .didChangeTheme,
-                                               object: nil)
+        tokenSet.registerOnUpdate(for: self) { [weak self] in
+            guard let strongSelf = self else {
+                return
+            }
+            strongSelf.updateColors()
```

Comment:
Why does only badge view follow the strong self pattern out of all our current controls here? (This would probably explain the massive delta compared to the others, right?)

Reply:
Hm, `NavigationBar` and `TableViewHeaderFooterView` both do as well, though of those only `NavigationBar` _needs_ to. I'll try removing it and see what happens.

Reply:
Oh sorry yea must have missed that! Although I guess my question still stands for why we need the strong self pattern (not to challenge you, just because I don't know and would like a better understanding!)

Reply:
You're right, we actually don't in any case! I thought that `NavigationBar` needed it because I didn't realize the argument to `updateColors` can be optional -- if it required a non-optional `UINavigationItem` then it would have to be strong, but I can optional-chain now instead.